mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBBasedQueryParserService.java

```diff
@@ -32,9 +31,16 @@
 
 	private volatile SessionFactoryEntityNamesResolver entityNamesResolver;
 
+	private ThreadLocal<QueryParser> parser = new ThreadLocal<QueryParser>() {
```

Comment:
Let's avoid this optimization for now. We will think about it later when we have some performance test